addons/crm/models/crm_team.py

```diff
@@ -78,3 +130,66 @@ def action_your_pipeline(self):
             ]
         action['context'] = action_context
         return action
+
+    def _compute_dashboard_button_name(self):
+        for team in self:
+            if team.use_opportunities:
+                team.dashboard_button_name = _("Pipeline")
+            else:
+                super(Team, team)._compute_dashboard_button_name()
```

Comment:
Rather always call super with `self` ; super method can be done it batch (one query for all record in self).
Calling it on only one record is not good for performance.
Your second proposition is way better :+1: